### PR TITLE
Y24-190-3 Milestone 2

### PR DESCRIPTION
**Pull request by sdjmchattie**
Closes https://github.com/sanger/sequencescape/issues/4244

#### Changes proposed in this pull request

- New endpoints added/updated:
    - BaitLibraryLayouts
    - PooledPlateCreations
    - SpecificTubeCreations
    - TagLayouts
- Tests updated to match.

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_

### CODE REVIEW

File: spec/requests/api/v2/transfers/transfers_spec.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 require 'rails_helper'
 require './spec/requests/api/v2/shared_examples/api_key_authenticatable'
-require './spec/requests/api/v2/shared_examples/invalid_post_requests'
+require './spec/requests/api/v2/shared_examples/post_requests'
```
[the remark this answers is not in the document]
Comment (sdjmchattie) [reply spec/requests/api/v2/transfers/transfers_spec.rb:6]: No longer specific to either.

File: app/models/bait_library_layout.rb [hunk 2]
```diff
@@ -46,46 +58,28 @@ def layout_bait_libraries_on_plate
...
-        yield(well, bait_library.first)
-      end
+  def first_bait_library(well)
+    bait_library =
+      well.aliquot_requests.for_submission_id(well.pool_id).map(&:request_metadata).map(&:bait_library).uniq
+
+    if bait_library.size > 1
+      raise StandardError,
+            "Multiple bait libraries found for #{well.map.description} on plate #{well.plate.human_barcode}"
+    end
```
[the remark this answers is not in the document]
Comment (sdjmchattie) [reply app/models/bait_library_layout.rb:69]: Yeah I agree but I think I should not increase the scope of my work as it's already so big.  This feels like something we should address in the future.

File: app/models/plate_creation.rb [hunk 3]
```diff
@@ -34,7 +34,7 @@ def children
     private :children
 
     def create_children!
-      self.child = child_purpose.create!(barcode: barcode)
+      self.child = child_purpose.create!(sanger_barcode: sanger_barcode)
```
Comment (stevieing) [inline app/models/plate_creation.rb:37]: `self.child = child_purpose.create!(sanger_barcode:)`

File: spec/resources/api/v2/transfer_template_resource_spec.rb [hunk 4]
```diff
@@ -8,28 +8,10 @@
...
-  it 'disallows updating of read-only fields', :aggregate_failures do
-    expect(resource).not_to have_updatable_field :uuid
-  end
+  # Attributes
+  it { is_expected.to have_readonly_attribute :uuid }
+  it { is_expected.to have_readwrite_attribute :name }
 
   # Filters
   it { is_expected.to filter(:uuid) }
-
```
Comment (stevieing) [inline spec/resources/api/v2/transfer_template_resource_spec.rb:29]: can this be removed?

**Review by stevieing — APPROVED** [review]

File: spec/resources/api/v2/transfer_template_resource_spec.rb (hunk 4 above)
Comment (sdjmchattie) [reply spec/resources/api/v2/transfer_template_resource_spec.rb:29]: The check for the filter?  The resource does contain the filter definition, so we'd have to remove that as well.  I don't think we should do that.

**Pull request merged**